accelerator/core/request/ta_send_mam.c

```diff
@@ -11,41 +11,67 @@
 ta_send_mam_req_t* send_mam_req_new() {
   ta_send_mam_req_t* req = (ta_send_mam_req_t*)malloc(sizeof(ta_send_mam_req_t));
   if (req) {
-    req->seed = NULL;
-    req->message = NULL;
-    req->channel_ord = 0;
-    req->ch_mss_depth = 6;
-    req->ep_mss_depth = 6;
-    req->psk = NULL;
-    req->ntru_pk = NULL;
+    req->data = NULL;
+    req->key = NULL;
   }
 
   return req;
 }
 
-void send_mam_req_free(ta_send_mam_req_t** req) {
-  if (!req || !(*req)) {
-    return;
+status_t send_mam_req_v1_init(ta_send_mam_req_t* req) {
+  if (req == NULL) {
```

Comment:
Why did you change the logic?

Reply:
We didn't introduce a interface for different interface, before this PR. 
The modification here is for the interface.